gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -0,0 +1,421 @@
+/**
+ * Copyright (C) 2019 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	tensor_filter_common.c
+ * @date	28 Aug 2019
+ * @brief	Common functions for various tensor_filters
+ * @see	  http://github.com/nnsuite/nnstreamer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @author	MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @bug	  No known bugs except for NYI items
+ *
+ */
+
+#include <string.h>
+
+#include <tensor_common.h>
+
+#include "tensor_filter_common.h"
+
+/**
+ * @brief Validate filter sub-plugin's data.
+ */
+gboolean
+nnstreamer_filter_validate (const GstTensorFilterFramework * tfsp)
+{
+  if (!tfsp || !tfsp->name) {
+    /* invalid fw name */
+    return FALSE;
+  }
+
+  if (!tfsp->invoke_NN) {
+    /* no invoke function */
+    return FALSE;
+  }
+
+  if (!(tfsp->getInputDimension && tfsp->getOutputDimension) &&
+      !tfsp->setInputDimension) {
+    /* no method to get tensor info */
+    return FALSE;
+  }
+
+  return TRUE;
+}
+
+/**
+ * @brief Filter's sub-plugin should call this function to register itself.
+ * @param[in] tfsp Tensor-Filter Sub-Plugin to be registered.
+ * @return TRUE if registered. FALSE is failed or duplicated.
+ */
+int
+nnstreamer_filter_probe (GstTensorFilterFramework * tfsp)
+{
+  g_return_val_if_fail (nnstreamer_filter_validate (tfsp), FALSE);
+  return register_subplugin (NNS_SUBPLUGIN_FILTER, tfsp->name, tfsp);
+}
+
+/**
+ * @brief Filter's sub-plugin may call this to unregister itself.
+ * @param[in] name The name of filter sub-plugin.
+ */
+void
+nnstreamer_filter_exit (const char *name)
+{
+  unregister_subplugin (NNS_SUBPLUGIN_FILTER, name);
+}
+
+/**
+ * @brief Find filter sub-plugin with the name.
+ * @param[in] name The name of filter sub-plugin.
+ * @return NULL if not found or the sub-plugin object has an error.
+ */
+const GstTensorFilterFramework *
+nnstreamer_filter_find (const char *name)
+{
+  return get_subplugin (NNS_SUBPLUGIN_FILTER, name);
+}
+
+
+/**
+ * @brief Parse the string of model
+ * @param[out] prop Struct containing the properties of the object
+ * @param[in] model_files the prediction model paths
+ * @return number of parsed model path
+ */
+guint
+gst_tensors_parse_modelpaths_string (GstTensorFilterProperties * prop,
```

Comment:
plz consider to change function name (gst_tensor_filter_xxx) later

Reply:
updated to `gst_tensor_filter_parse_modelpaths_string`